apps/general/models.py

```diff
@@ -1,3 +1,230 @@
 from django.db import models
+from django.core.validators import RegexValidator
+from passlib.hash import bcrypt 
+import bcrypt,uuid,base64
 
-# Create your models here.
+def company_logos(instance, filename):
+    company_name = instance.name.replace(' ', '_')
+    unique_id = uuid.uuid4().hex[:6] #Generates a unique identifier
+    return f"company_logos/{company_name}_{unique_id}/{filename}"
+
+class Companies(models.Model):
+    company_id = models.AutoField(primary_key=True)
+    name = models.CharField(max_length=255, null=False)
+    print_name = models.CharField(max_length=255, null=True, default=None)
+    short_name = models.CharField(max_length=100, null=True, default =None)
+    code = models.CharField(max_length=100, null=True, default=None)
+    num_branches = models.IntegerField(default=0)
+    logo = models.ImageField(null=True, upload_to=company_logos, default=None)
+    address = models.TextField(null=True)
+    country = models.CharField(max_length=100,null=True, default=None)
+    state = models.CharField(max_length=255, null=True, default=None)
+    city = models.CharField(max_length=255, null=True, default=None)
```

Comment:
Companies table doesn't have country and state columns, it has only city_id and it is Integer datatype

Reply:
Yeah sorry Thulasi. Confused as branches having these columns by mistakenly I have mentioned here also.I will change it Thulasi.